src/dotnet/ToolPackage/IProjectRestorer.cs

```diff
@@ -11,6 +12,7 @@ internal interface IProjectRestorer
             FilePath projectPath,
             DirectoryPath assetJsonOutput, 
             FilePath? nugetconfig,
-            string source);
+            string source,
+            IEnumerable<string> restoreArguments);
```

Comment:
Either merge all the param to restoreArguments including source, nuget config. Or explicit about what restore arguments allow to pass.

Reply:
Yeah, I was looking for a better way to do this.  The problem is that I didn't want to have to define a `verbosity` option into this command because it is currently implemented as a "forwarded argument".  Thus I chose to simply pass through the forwarded arguments this way rather than taking a single option for "verbosity".  I'd be happy to do it the other way and just define the option manually rather than using `CommonOptions.VerbosityOption` if you think that's best.

Reply:
I think "CommonOptions.VerbosityOption" is better. restoreArguments feel like the caller can passing a lot of opinions. But a string array cannot compose